pkg/agents/team4/finalAgent/trustScoreUpdates.go

```diff
@@ -0,0 +1,117 @@
+							}
+						}
+						break
+					}
+				}
+
+			}
+
+		}
+	} else if a.params.lastFoodTaken+a.CurrPlatFood() != a.params.lastPlatFood {
+		for i := 0; i < len(a.params.responseMessages); i++ { // Iterate through each response message
+			respMsg := a.params.responseMessages[i]
+			resMsg, ok := respMsg.(messages.ResponseMessage)
+			if !ok {
+				err := fmt.Errorf("ResponseMessage type assertion failed")
+				fmt.Println(err.Error())
+			} else {
+				for j := 0; j < len(a.params.sentMessages); j++ { // Iterate through each sent message
+					sentMsg := a.params.sentMessages[j]
+
+					if resMsg.RequestID() == sentMsg.ID() { // Find the corresponding response message that's been sent
+						a.params.sentMessages = remove(a.params.sentMessages, j) // Remove the accessed response/sent messages from memory
+						a.params.responseMessages = remove(a.params.responseMessages, i)
+						a.Log("Team4 received a message", infra.Fields{"sender_uuid": msg.ID(), "sentmessage_uuid": sentMsg.ID()})
+
+						if sentMsg.MessageType() == messages.RequestTakeFood && a.Floor()-msg.SenderFloor() == -1 {
+							reqMsg, ok := sentMsg.(messages.RequestMessage)
+							if !ok {
+								err := fmt.Errorf("RequestMessage type assertion failed")
+								fmt.Println(err.Error())
+							} else if food.FoodType(reqMsg.Request()) >= a.NeighbourFoodEaten() {
+								a.Log("Team4: For Requested Food to Take greater then or equal neighbour food eaten", infra.Fields{"Request_amt": reqMsg.Request(), "Food_on_our_level": a.NeighbourFoodEaten(), "global_trust": a.params.globalTrust})
+								a.AddToGlobalTrust(a.params.coefficients[1])
+							} else if food.FoodType(reqMsg.Request()) < a.NeighbourFoodEaten() {
+								a.Log("Team4: For Requested Food to Take less than neighbour food eaten", infra.Fields{"Request_amt": reqMsg.Request(), "Food_on_our_level": a.NeighbourFoodEaten(), "global_trust": a.params.globalTrust})
+								a.SubFromGlobalTrust(a.params.coefficients[1])
+							}
+						}
+						break
+					}
+				}
+			}
+		}
+
+	} else {
+		for j := 0; j < len(a.params.sentMessages); j++ {
+			if msg.RequestID() == a.params.sentMessages[j].ID() {
+				sentMsg := a.params.sentMessages[j]
+				a.Log("Team4 received a message", infra.Fields{"sender_uuid": msg.ID(), "sentmessage_uuid": sentMsg.ID()})
+
+				if sentMsg.MessageType() == messages.RequestTakeFood && a.NeighbourFoodEaten() == -1 {
+					a.params.sentMessages = append(a.params.responseMessages, &msg)
+				} else if sentMsg.MessageType() == messages.RequestLeaveFood && !a.PlatformOnFloor() {
+					a.params.sentMessages = append(a.params.responseMessages, &msg)
+				} else if sentMsg.MessageType() == messages.RequestTakeFood && a.Floor()-msg.SenderFloor() == 1 {
+					a.AddToGlobalTrust(a.params.coefficients[1])
+				} else if sentMsg.MessageType() == messages.RequestLeaveFood && a.Floor()-msg.SenderFloor() == -1 {
+					a.AddToGlobalTrust(a.params.coefficients[1])
+				}
+				break
+			}
+		}
+	}
+}
+
```

Comment:
I'm not even going to bother reading this code until:

- You make it shorter
- You get rid of all of these nested if statements
- You put things into functions so it's easy to tell what each section does.